website/conferences/views.py

```diff
@@ -180,14 +184,122 @@ def _render_conference_node(node, idx, conf):
         'tags': ' '.join(tags)
     }
 
+def filter_and_sort_conference_data(nodes, conf):
+    """
+    Filter and sort conference submissions
+    Returns: filtered/sorted node queryset.  Sorts on download count by default.
+
+    :param obj nodes: Node queryset - conference submissions
+    :param conf obj: Conference
+    """
+    q = request.args.get('q', '')
+    # Give "sort" a default value, since pagination may be inconsistent with unordered objects.
+    sort = request.args.get('sort', '-downloads')
+
+    if q:
+        format_q = '%' + q + '%'
+        # TODO replace this raw sql with a regular django query when django-include
+        # limitations are fixed: Subqueries can only return one column, but a first visible
+        # contributor subquery also fetches guids, despite .include(None)
+        # This subquery looks for "q" query param in the node title and the first visible contrib's fullname.
+        raw_queryset = nodes.raw(
+            """
+            SELECT *
```

Comment:
good tip, I looked into this but couldn't get `SELECT id` to work. This results in a `refresh_from_db() got an unexpected keyword argument 'fields'` error.  I think this is because I am using raw sql (which I used instead of executing custom sql directly because I further wanted to restrict the nodes queryset, otherwise I'd have to do the whole nodes piece in sql too).

If instead I execute custom sql directly, I *can* get an array of ids returned here, using array_agg, but would rather avoid this because of my reason above.  I just couldn't get the raw queryset to return one column.